### ISSUE TITLE
eigen_spatial_convolutions.h: reading of uninitialized variable 'k'

### ISSUE THREAD

**Issue by gavinsherry** [tensorflow/tensorflow#2877]
It appears that `k` is not initialized before being read below. From eigen_spatial_convolutions.h:848:

```
       for (Index k; k < depth; k++) {
          block[0] = dm0.loadCoeffStandard(k);
          block[1] = dm1.loadCoeffStandard(k);
          block[2] = dm2.loadCoeffStandard(k);
          block[3] = dm3.loadCoeffStandard(k);
          block += 4;
        }
      } else {
        for (Index k; k < depth; k++) {
          block[0] = dm0(k);
...
```

Am I missing something here?

**Comment by gavinsherry** [comment]
@benoitsteiner seems like you fixed this in 406db8835d656540c96f69e9ac565ef8c22be3d1. This can be closed now.

**Comment by benoitsteiner** [comment]
Indeed
